src/runner_manager.py

```diff
@@ -685,42 +595,27 @@ def create_runner_info(
 
         return runners
 
-    def _get_github_registration_token(self) -> str:
-        """Get token from GitHub used for registering runners.
-
-        Returns:
-            The registration token.
-        """
-        token: RegistrationToken
-        if isinstance(self.config.path, GitHubRepo):
-            token = self._clients.github.actions.create_registration_token_for_repo(
-                owner=self.config.path.owner, repo=self.config.path.repo
-            )
-        elif isinstance(self.config.path, GitHubOrg):
-            token = self._clients.github.actions.create_registration_token_for_org(
-                org=self.config.path.org
-            )
-        else:
-            assert_never(token)
-
-        return token["token"]
+    def build_runner_image(self) -> None:
+        """Build the LXD image for hosting runner.
 
-    def _get_github_remove_token(self) -> str:
-        """Get token from GitHub used for removing runners.
+        Build container image in test mode, else virtual machine image.
 
-        Returns:
-            The removing token.
+        Raises:
+            LxdError: Unable to build the LXD image.
         """
-        token: RemoveToken
-        if isinstance(self.config.path, GitHubRepo):
-            token = self._clients.github.actions.create_remove_token_for_repo(
-                owner=self.config.path.owner, repo=self.config.path.repo
-            )
-        elif isinstance(self.config.path, GitHubOrg):
-            token = self._clients.github.actions.create_remove_token_for_org(
-                org=self.config.path.org
-            )
-        else:
-            assert_never(token)
-
-        return token["token"]
+        cmd = ["/usr/bin/bash", BUILD_IMAGE_SCRIPT_FILENAME]
+        if LXD_PROFILE_YAML.exists():
+            cmd += ["test"]
+        execute_command(cmd)
+
+    def schedule_build_runner_image(self) -> None:
+        """Install cron job for building runner image."""
+        cron_file = self.cron_path / "build-runner-image"
+        # Randomized the time executing the building of image.
```

Comment:
Perhaps include why, I'm guessing so that all the runners don't do it at the same time and end up getting rate limited?

Reply:
Yes. The charm hardcodes the time, then all instance of the charm will try to build the image at the same time, which will cause a spike in disk, memory, and network usage for a server that has multiple deployments.